Kernel/FileSystem/BlockBasedFileSystem.cpp

```diff
@@ -116,6 +124,15 @@ BlockBasedFileSystem::BlockBasedFileSystem(OpenFileDescription& file_description
 
 BlockBasedFileSystem::~BlockBasedFileSystem() = default;
 
+void BlockBasedFileSystem::remove_disk_cache_before_last_unmount()
+{
+    VERIFY(m_lock.is_locked());
+    m_cache.with_exclusive([&](auto& cache) {
+        cache->clean_before_clear();
```

Comment:
Why shouldn't we do this clean_before_clear behavior in the destructor of the DiskCache? If there's an ordering issue of destructors, we should re-order the members of DiskCache so that they are cleared in the right order